core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -588,7 +588,7 @@ private[spark] class SparkSubmit extends Logging {
       OptionAssigner(args.deployMode, ALL_CLUSTER_MGRS, ALL_DEPLOY_MODES,
         confKey = SUBMIT_DEPLOY_MODE.key),
       OptionAssigner(args.name, ALL_CLUSTER_MGRS, ALL_DEPLOY_MODES, confKey = "spark.app.name"),
-      OptionAssigner(args.ivyRepoPath, ALL_CLUSTER_MGRS, CLIENT, confKey = "spark.jars.ivy"),
+      OptionAssignerWrapper(args.ivyRepoPath, ALL_CLUSTER_MGRS, CLIENT, confKey = "spark.jars.ivy"),
```

Comment:
Seems fine either way honestly.
My main concern is consistency - why are these options handled differently from those using OptionAssigner?

Reply:
> Seems fine either way honestly.
> My main concern is consistency - why are these options handled differently from those using OptionAssigner?

Since I want to add a constructor to support Option[String]. Emmm seems will cause `None.get` exception.
Change to to use `orNull`

Reply:
I see, OptionAssigner takes an object or null, not an Option. Ok either way. It's fine as is now.